### ISSUE TITLE
Provide a TSX string scope

### ISSUE THREAD

**Issue by orta** [Microsoft/TypeScript-TmLanguage#457]
**[TS and JS Grammar Extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode.typescript-javascript-grammar) version:**  0.0.20 / latest

**Code**

```ts
const thing = () =>
    <ScrollView style={{backgroundColor: "black"}}>
          <LargeHeadline>Sell works from your collection through our partner network</LargeHeadline>
    </ScrollView>
```

and 

```ts
export default class Info extends React.Component<Props, any> {
  render() {
    return (<ScrollView style={{backgroundColor: "black"}}>
          <LargeHeadline>Sell works from your collection through our partner network</LargeHeadline>
    </ScrollView>)
  }
}
```

End up rendering differently for the strings elements as children e.g.`"Sell works from your collection through our partner network "`. This is OK, as the top one falls through to `source,js` and the latter goes to `meta.class.js`

![screen shot 2017-05-22 at 11 49 56](https://cloud.githubusercontent.com/assets/49038/26304922/1751e09a-3ee5-11e7-80cb-f95e1132e34a.png)

It would be awesome if there was a scope for strings inside JSX elements that represents a string for text. Perhaps something like `meta.jsx.string`?

**Comment by sheetalkamat** [comment]
I don't understand the issue here. The both code samples you have given colorize and tokenize exactly same:
![jsx1](https://cloud.githubusercontent.com/assets/8052792/26421380/bd793770-407a-11e7-834c-9b4d4e1d71e5.JPG)
![jsx2](https://cloud.githubusercontent.com/assets/8052792/26421390/c5b6de56-407a-11e7-9d85-5309ec02739a.JPG)

**Comment by orta** [comment]
Looks like this is theme specific. In the color theme [I'm using](https://github.com/smlombardi/theme-slime), they specifically have a color for `meta.class` instead of just falling back to `source.js`.

When I checked out all the other themes I have installed, this was the only one that had had this issue. So this is probably a non-issue. It would be a nice-to-have, but unless it's on the roadmap for other reasons - I'm gonna close this.
